### ISSUE TITLE
Don't use "unsigned" by itself as a type

### ISSUE THREAD

**Issue by evancox10** [riscv-software-src/riscv-isa-sim#138]
In debug_module.cc and debug_module.h, there are a lot of usages of `unsigned` by itself as a type. According to the C++ specs, this is apparently the same as `unsigned int`, however this usage is not very common and it will likely confuse a lot of people. Also, it is not portable across architectures, C++ does not guarantee that `unsigned int` is 32 bits, only that it is at least 16. Here, we explicitly want a 32-bit data type to represent the Debug spec's 32-bit data structures & address space. 

I recommend code like the following:
```
// debug_module.h:
  // Debug Module Interface provides a 32-bit address space
  typedef uint32_t debug_addr_t
  // Debug Module Interface provides a 32-bit data bus
  typedef uint32_t debug_data_t
```

```
// debug_module.cc
    bool dmi_read(debug_addr_t address, debug_data_t *value);
```

I'll try to do a PR if I have time, but no guarantees

**Comment by timsifive** [comment]
I agree it's better to be clear about what size is actually required. abits in the simulated JTAG module is 6, so uint8_t suffices there. I'll get to this next week, probably.
